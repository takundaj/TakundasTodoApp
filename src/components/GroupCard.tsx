import {
  View,
  Text,
  StyleSheet,
  Touchable,
  TouchableOpacity,
  Dimensions,
} from "react-native";
import React from "react";
import Ionicons from "@expo/vector-icons/Ionicons";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { StackParamList } from "../types/navigationTypes";

type Props = {};

type GroupCardNavigationProp = NativeStackNavigationProp<StackParamList>;

const GroupCard = (props: Props) => {
  const navigation = useNavigation<GroupCardNavigationProp>();

  const handleOnPress = () => {
    navigation.navigate("TodosScreen");
  };

  return (
    <TouchableOpacity style={styles.container} onPress={handleOnPress}>
      <View style={styles.innerContainer}>
        <Ionicons name="list-outline" size={32} color="green" />
        <View>
          <Text style={styles.groupTitle}>All</Text>
          <Text style={styles.todoCount}>23 Tasks</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export const AddGroupCard = (props: Props) => {
  const navigation = useNavigation<GroupCardNavigationProp>();

  const handleOnPress = () => {
    navigation.navigate("TodosScreen");
  };

  return (
    <TouchableOpacity style={styles.addGroupContainer} onPress={handleOnPress}>
      <Ionicons name="add-outline" size={64} color="green" />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    width: Dimensions.get("screen").width / 2.6,
    height: 150,
    padding: 16,
    margin: 16,
    borderWidth: 1,
    borderRadius: 8,
    backgroundColor: "white",
  },
  innerContainer: {
    flex: 1,
    borderLeftWidth: 4,
    borderColor: "turquoise",
    paddingHorizontal: 16,
    paddingBottom: 8,
    justifyContent: "space-between",
  },
  groupTitle: {
    fontSize: 16,
    marginBottom: 4,
  },
  todoCount: {
    color: "grey",
  },
  addGroupContainer: {
    width: 150,
    height: 150,
    padding: 16,
    margin: 16,
    borderWidth: 1,
    borderRadius: 8,
    backgroundColor: "white",
    alignItems: "center",
    justifyContent: "center",
  },
});

export default GroupCard;
