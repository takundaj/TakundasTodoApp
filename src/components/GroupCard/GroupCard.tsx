import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
} from "react-native";
import React from "react";
import Ionicons from "@expo/vector-icons/Ionicons";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { StackParamList } from "../../types/navigationTypes";
import styles from "./groupCard.style";

type Props = {
  title?: string;
  groupId?: string;
  groupColor: string;
  numberOfTodos?: number;
};

type GroupCardNavigationProp = NativeStackNavigationProp<StackParamList>;

const GroupCard = ({ title, numberOfTodos, groupId, groupColor }: Props) => {
  const navigation = useNavigation<GroupCardNavigationProp>();

  const handleOnPress = () => {
    navigation.navigate("TodosScreen", {
      groupTitle: title,
      groupId,
      groupColor,
    });
  };

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={handleOnPress}
      onLongPress={() =>
        navigation.navigate("GroupModal", {
          groupTitle: title,
          groupColor,
          groupId,
        })
      }
      testID="group-card-container"
    >
      <View
        style={[
          styles.innerContainer,
          { borderColor: groupColor || "turquoise" },
        ]}
      >
        <Ionicons
          name="list-outline"
          size={32}
          color={groupColor || "turquoise"}
        />
        <View>
          <Text style={styles.groupTitle}>{title}</Text>
          <Text style={styles.todoCount}>{numberOfTodos} Tasks</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default GroupCard;
