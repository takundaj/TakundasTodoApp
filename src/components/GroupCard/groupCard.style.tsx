import { StyleSheet, Dimensions } from "react-native";
import { size } from "../../utils/helpers";

export default StyleSheet.create({
  container: {
    width: Dimensions.get("screen").width / 2.6,
    height: 150,
    padding: size(4),
    margin: size(4),
    borderRadius: size(2),
    backgroundColor: "white",
    shadowOffset: {
      width: 0.5,
      height: 1,
    },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
  innerContainer: {
    flex: 1,
    borderLeftWidth: size(1),
    paddingHorizontal: size(4),
    paddingBottom: size(2),
    justifyContent: "space-between",
  },
  groupTitle: {
    fontSize: size(4),
    marginBottom: size(1),
  },
  todoCount: {
    color: "grey",
  },
  addGroupContainer: {
    width: 150,
    height: 150,
    padding: size(4),
    margin: size(4),
    borderWidth: 1,
    borderRadius: size(2),
    backgroundColor: "white",
    alignItems: "center",
    justifyContent: "center",
  },
});
