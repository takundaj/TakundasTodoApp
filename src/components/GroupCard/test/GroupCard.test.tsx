import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import GroupCard from "../GroupCard";
import { useNavigation } from "@react-navigation/native";

const mockProps = {
  title: "dummyTitle",
  numberOfTodos: 5,
  groupId: "dummyId",
  groupColor: "red",
};

jest.mock("@react-navigation/core", () => ({
  ...jest.requireActual("@react-navigation/core"),
  useNavigation: jest.fn(),
}));

test("Should render GroupCard component", () => {
  const { getByTestId } = render(<GroupCard {...mockProps} />);

  const container = getByTestId("group-card-container");
  expect(container).toBeDefined();
});

test("Should match snapshot", () => {
  const tree = renderer.create(<GroupCard {...mockProps} />);
  expect(tree).toMatchSnapshot();
});
