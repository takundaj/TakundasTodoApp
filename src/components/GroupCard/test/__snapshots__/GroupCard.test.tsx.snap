// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 8,
      "height": 150,
      "margin": 16,
      "opacity": 1,
      "padding": 16,
      "shadowOffset": {
        "height": 1,
        "width": 0.5,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
      "width": 288.46153846153845,
    }
  }
  testID="group-card-container"
>
  <View
    style={
      [
        {
          "borderLeftWidth": 4,
          "flex": 1,
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingHorizontal": 16,
        },
        {
          "borderColor": "red",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "red",
            "fontSize": 32,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ÔçΩ
    </Text>
    <View>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 4,
          }
        }
      >
        dummyTitle
      </Text>
      <Text
        style={
          {
            "color": "grey",
          }
        }
      >
        5
         Tasks
      </Text>
    </View>
  </View>
</View>
`;
